staging/src/k8s.io/apiserver/pkg/apis/config/types.go

```diff
@@ -89,4 +89,22 @@ type KMSConfiguration struct {
 	// Timeout for gRPC calls to kms-plugin (ex. 5s). The default is 3 seconds.
 	// +optional
 	Timeout *metav1.Duration
+	// HealthzEndpoint contains information required to validate healthz of kms-plugin.
+	// +optional
+	HealthzEndpoint *HealthzEndpoint
+}
+
+// HealthzEndpoint contains information required to validate healthz of kms-plugin.
+type HealthzEndpoint struct {
```

Comment:
have you thought about how (and whether) the results from this would be incorporated into the `/healthz` endpoint for the API server? Do we need any of the threshold options used for readiness/liveness probes?
https://github.com/kubernetes/kubernetes/blob/d1e828f8dfc77d1a6a8d7b0bd8ff0006f058b8fe/staging/src/k8s.io/api/core/v1/types.go#L2009-L2035

Reply:
It's not clear from these docs that this is only used on startup. my initial expectation was that this would factor into apiserver `/healthz` as well (other comments/questions were based on that assumption). If the kms goes unhealthy after start, does anything notice that and flag the apiserver as unhealthy currently?

Reply:
1. I think it would be great to be able to incorporate this into /healthz of kube-apiserver. Frankly, I was not able to find an example of how to do this. Could you point me to something I could use as an example.

2. I think I may have re-invented the wheel with HealthzEndpoint; Probe would have done nicely. Should I refactor to Probe? 

3. I fixed the docs by making it clear that this check only takes place during initialization of kube-apiserver.

4. When kms-plugin goes unhealthy after kube-apiserver is started, (to the best of my knowledge) this fact is not reflected in the /healthz of the kube-apiserver. 
However, we have a metric that gets incremented every time we are unable to transform - this is how one could implement alerts for this condition. Of course, writes against Secrets are no longer possible which is also reflected in some counters at the REST layer.

Reply:
What do you think about throwing this into a [postStartHook](https://github.com/kubernetes/kubernetes/blob/83f6b8869f4430d9698077ea26f98d3fbadb17e7/staging/src/k8s.io/apiserver/pkg/server/hooks.go#L75-L106) @liggitt? It feels boot-sequence-ish to me.